### PR TITLE
[Bug] Shopping list does not shown / bug#2206

### PR DESCRIPTION
**Pull request by yuriiLisunov**
**Description:**
Loader is displayed on left panel on 'My habits' page instead shopping list under 'My eco friends'. 
BEFORE:
BEFORE
AFTER:
'Shopping list' section with correspond items is displayed on left panel on 'My Habits' page.
after

### CODE REVIEW

**Review by VitalyBorysenko — APPROVED** [review]

**Review by idzhersh — APPROVED** [review]

**Review by paslavskyi9roman — APPROVED** [review]

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts [hunk 1]
```diff
@@ -20,30 +20,26 @@ export class ShoppingListComponent implements OnInit {
 
   public getShoppingList(): void {
     this.profileSubscription = this.profileService.getShoppingList()
-      .subscribe((success: ShoppingList[]) => this.shoppingList = success);
+    .subscribe(
```
Comment (mr-pavlenko) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.ts:23]: Use subscription instead of the flag.

**Review by KirillAbramo — APPROVED** [review]

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts (hunk 1 above)
Comment (yuriiLisunov) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.ts:22]: fixed

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts [hunk 2]
```diff
@@ -20,30 +19,20 @@ export class ShoppingListComponent implements OnInit {
...
+      (shoppingListArr: ShoppingList[]) => this.shoppingList = shoppingListArr,
+      error => this.shoppingList = []
+    );
   }
 
   public toggleDone(item): void {
     this.profileService.toggleStatusOfShoppingItem(item)
-      .subscribe((success) => this.updateDataOnUi(item));
+    .subscribe((success) => this.updateDataOnUi(item));
```
Comment (mr-pavlenko) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.ts:30]: Need to unsubscribe.

Comment (yuriiLisunov) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.ts:33]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.html [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
+            <h2 class="header">
+                {{ 'profile.shoppingList' | translate }} <span>{{ shoppingList?.length }}</span></h2>
+            <a class="btn-see-all">{{ 'profile.see-all' | translate }}</a>
+        </div>
+        <div *ngIf="profileSubscription && !shoppingList">
+                <mat-spinner color="grey" 
+                diameter="70" 
+                class="loading"></mat-spinner>
+        </div>
+        <div *ngIf="profileSubscription && shoppingList && shoppingList.length>0">
```
[the remark this answers is not in the document]
Comment (yuriiLisunov) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.html]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.html [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
+                        <mat-checkbox [checked]='shoppingItem.status === "DONE"' 
+                                        class="checkbox" 
+                                        color="primary"
+                                        (click)="toggleDone(shoppingItem)"
+                        ></mat-checkbox>
+                    </li>
+                </ul>
+            </div>
+        </div>
+        <div *ngIf="profileSubscription && shoppingList && shoppingList.length===0">
```
[the remark this answers is not in the document]
Comment (yuriiLisunov) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.html]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.scss [hunk 5]
```diff
@@ -1,9 +1,23 @@
+@import url('https://fonts.googleapis.com/css?family=Montserrat&display=swap');
```
[the remark this answers is not in the document]
Comment (yuriiLisunov) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.scss]: fixed

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.scss [hunk 6]
```diff
@@ -19,13 +33,21 @@ $bp-largest: 1440px;
...
 
 h2 > span {
   color: var(--tertiary-light-grey);
-  margin-left: 10px;
+}
+
+.empty-list {
+  text-align: center;
+  font-size: 12px;
+  font-family: 'Montserrat', sans-serif;
```
[the remark this answers is not in the document]
Comment (yuriiLisunov) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.scss]: fixed

**Review by AlekOK — APPROVED** [review]

**Review by mr-pavlenko — APPROVED** [review]

**Pull request merged**